### PR TITLE
handle Proc in if options

### PR DESCRIPTION
**Pull request by hallgren**
The current spec I wrote did not check if the `if` option were a Proc.

### CODE REVIEW

File: yaks/lib/yaks/mapper/attribute.rb [hunk 1]
```diff
@@ -2,22 +2,23 @@ module Yaks
...
-      include Util
-      include Attribs.new(:name, :block, options: {}.freeze)
+      include Attribs.new(:name, :block, if: true), Util
 
-      def self.create(*args, &block)
-        args, options = extract_options(args)
-        new(name: args.first, options: options, block: block)
+      def self.create(name, options = {}, &block)
+        options[:name] = name
+        options[:block] = block
```
Comment (plexus) [inline yaks/lib/yaks/mapper/attribute.rb]: You're mutating an object passed in by the caller. That's not a good idea. Better to use something like

``` ruby
options = options.merge(...)
```

File: yaks/spec/unit/yaks/mapper/attribute_spec.rb [hunk 2]
```diff
@@ -16,7 +15,7 @@
     its(:block) { should be_nil }
 
     it "should accept two parameter" do
-      expect{described_class.create(:the_name, {})}.not_to raise_error()
+      expect{described_class.create(:the_name)}.not_to raise_error()
```
Comment (groyoh) [inline yaks/spec/unit/yaks/mapper/attribute_spec.rb]: The code here does not match the example name anymore.

File: yaks/lib/yaks/mapper/attribute.rb [hunk 3]
```diff
@@ -2,22 +2,23 @@ module Yaks
...
+        new(options)
       end
 
       def add_to_resource(resource, mapper, _context)
-        return unless mapper.expand_value(options.fetch(:if, true))
-
+        return resource unless Resolve(self.if, mapper)
```
Comment (groyoh) [inline yaks/lib/yaks/mapper/attribute.rb]: I think there should be another spec to test the proc/lambda is actually `instance_eval/exec` on the mapper.

File: yaks/spec/unit/yaks/mapper/attribute_spec.rb [hunk 4]
```diff
@@ -70,11 +72,41 @@
   end
 
   describe "#add_to_resource" do
+
+    let(:options) { {if: ->{ true }} }
+    let(:options_false) { {if: ->{ 0 == 1 }} }
```
Comment (hallgren) [inline yaks/spec/unit/yaks/mapper/attribute_spec.rb]: @groyoh I think this lines does that.

Comment (groyoh) [inline yaks/spec/unit/yaks/mapper/attribute_spec.rb]: It checks that the proc is `call`'ed but not `instance_eval`'d. What I mean is to have something like:

```ruby
let(:options_false) { {if: ->{ object.name.length > 10 }} }
```

**Comment by plexus** [comment]
>  don´t know that I'm doing but seams to work.. 

:smile: is that referring to the use of `Resolve`?

Jest aside, please pay some more attention to commit messages. I know I don't always do this right either, but have a look at the history and you will find many good examples. The typical format is a single line of less than 70 or 80 characters, then a blank line, then a full explanation, manually line wrapped at 80 chars.

A small tip when working on pull requests: I don't get any notifications when you push new code. If you're done making changes (e.g. you've addressed the review comments), it's a good idea to leave a comment here briefly summarising what you changed.

That said I think code wise this is good enough to go in. Maybe you can squash the commits together into one and give it a more elaborate description? I can put out a new release afterwards.

**Comment by hallgren** [comment]
Thank for the tips, commiting code on a fever seams like a bad idea.. Looks like i didn´t knew what I were doing...

I try to rap this one up the coming week.

**Pull request closed without merging**